dl_bt_metaquotes.py

```diff
@@ -233,30 +234,37 @@ def decompress(data, year, month):
         #   Description : kind of repeater type for highly compressable one minute step data
         #   Block length: 1 byte flag + streak*6 bytes of data
         #   Flag codes  : 0x8a..0xbf stores the length of streak in this type of block
-        #   Field_1     : TODO
-        #   Field_2     : TODO
-        #   Field_3     : TODO
-        #   Field_4     : TODO
-        #   Field_5     : TODO
-        #   Field_6     : TODO
+        #   Field_1     : timestamp incremental to previous timestamp
+        #   Field_2     : 1 byte, open price incremental stored as a signed char and added to previous close price
+        #   Field_3     : 1 byte, high price incremental stored as an unsigned char and added to open price
+        #   Field_4     : 4 bytes, low price incremental stored as an unsigned short subtracted from open price
+        #   Field_5     : 4 bytes, close price incremental stored as an unsigned short and added to open price
+        #   Field_6     : 1 byte, volume incremental stored as a signed char
         elif data[i] > 0x7F:
             streak = (
                 data[i] - 0x7F
             )  # Get the number of streaks hiding inside this block
-            i += 1  # Move index to first data byte of current streak
+            i += 1  # Move index to first data byte of current streak            
+
+            timestamp = lastBar["timestamp"] + datetime.timedelta(minutes=1)
+
+            open      = lastBar["close"] + unpack("b", bytes([data[i]]))[0]
+            high      = open + unpack("B", bytes([data[i + 1]]))[0]
```

Comment:
The `high` value being decoded as 1 byte, and `low` as 4 bytes, doesn't make much logical sense I think. I think something is still not right.

Reply:
What I see we have now:
- `open` is: `last close` + `b` = signed char (1 byte), byte `i + 0`
- `high` is: `open` + `B` = unsigned char (1 byte), byte `i + 1`
- `low` is: `open` - `<H` = unsigned short (2 bytes), byte `i + 8` and `i + 9`
- `close` is: `open` + `<H` = unsigned short (2 bytes), byte `i + 10` and `i + 11`
- `volume` is: `B` = unsigned char (1 byte), byte `i + 9`

My conclusion:
- `high` will be always equal or greater than `open` as we adds unsigned char - which is good
- `low` will be always equal or less than `open` as we subtracts unsigned short - which is good
- `close` will never be less that `open` (but it should be able to), because we adds `unsigned short` - looks like we need to use little-endian signed short - `<h` instead of `<H`, so `close` could be less, equal or greater than `open`
- `volume` uses byte `i + 9` which was used as second byte for `low` - looks like checking wrong byte index or maybe `low` should use single byte, the same as `high`

Reply:
Hi, I'll make some updates and send it again to review.

Reply:
Hi all,
I was doing some testing and comparing the existing data from https://github.com/FX-Data/FX-Data-EURUSD-DS/tree/EURUSD-2022/EURUSD/2022/02 and https://github.com/FX-Data/FX-Data-EURUSD-DS/tree/EURUSD-2022/EURUSD/2022/03 and I believe now that the results are quite more consistent than it was on the previous review.

So, I changed the code to use the same code presented on the Type-3 block but using a different streak.

Can you please review it when you got a chance? Thanks.